R/join_keys.R

```diff
@@ -142,6 +142,7 @@ join_keys.teal_data <- function(...) {
 #' join_keys(td)
 `join_keys<-.teal_data` <- function(x, value) {
   join_keys(x@join_keys) <- value
+  x@datanames <- sort_datanames(x@datanames, x@join_keys)
```

Comment:
better if `datanames()` getter do the sort

```suggestion
```

Reply:
I think it's better if `x@datanames` has the same order as `datanames(x)` every time. That's why I am overwriting the filed in here.

If there is already a field `x@datanames`, and someone sets `join_keys(x)` then if we don't update the filed and make the getter return the order, then 

`x@datanames` can be different than `datanames(x)`

Reply:
> x@datanames can be different than datanames(x)

fair enough 👍